### ISSUE TITLE
Checkboxes in parameter dialog with dependency to string choices should be updated immediately

### ISSUE THREAD

**Issue by niklwors** [OpenModelica/OpenModelica#10145]
### Description

If a Boolean parameter should only be editable in the parameter dialog depending on a previously selected string choice, this is not possible in the OMEdit parameter dialog and the new instantiation-based  UI. You have to close the parameter dialog first and open it again to do so.
e.g 
```modelica
parameter String controller = "Velocity controller" 
  annotation( choices(
      choice="Torque controller",
      choice="Velocity controller",
      choice="Position controller"), Dialog(group="Control Selection"));

 parameter Boolean useAdditiveVelocity=false  annotation( Dialog(
  enable=controller =="Position controller",
  group="Control Selection"),choices(checkBox=true));
```
picture 1:
<img width="579" alt="StringChoiceProblem1" src="https://user-images.githubusercontent.com/12269776/216381659-1ad28c82-8f34-41a0-a3f4-be8a9588a39d.png">

You need to close the parameter dialog first before it is possible
picture 2:
<img width="594" alt="StringChoiceProblem2" src="https://user-images.githubusercontent.com/12269776/216381679-e3bbe6c0-4656-4362-b862-7e60d9c6ce07.png">

### Steps to Reproduce
Load the package ```Rexroth_ElectricDrives/package.mo```
Drag and drop the component ```Rexroth_ElectricDrives.GenericRotationalDrive``` onto a new sheet.
Open the paramter dialog and select ```position controller``` from the controller menu

### Expected Behavior

After selection the position controller the fields from picture 1 are active.

### Version and OS

 - OpenModelica Version: v1.21.0-dev-215-gfbc4a47daf
 - OS: Windows 10, 64 bit
 - MSL 4.0.0
 - BRSL 2.2.1

**Comment by adeas31** [comment]
PR #10187 partially fixes the issue. Now changing the `controller` value updates the `checkboxes` but does not update the text fields. The text fields depends on `controlMode` parameter which is evaluated by the frontend. I talked with @perost about it and he will further investigate this.

**Comment by perost** [comment]
> PR #10187 partially fixes the issue. Now changing the `controller` value updates the `checkboxes` but does not update the text fields. The text fields depends on `controlMode` parameter which is evaluated by the frontend. I talked with @perost about it and he will further investigate this.

I opened #10188 for that issue.

**Comment by casella** [comment]
@niklwors, PR #10187 was merged in, and 10188 was also fixed, so this issue should be fixed now.

Please reopen you still see problems. Thanks!

**Comment by casella** [comment]
@perost can you please check?

**Comment by abuntrock** [comment]
Small addition here. This is also not possible if the enable statement depends on the value of another Real parameter

**Comment by sbaetzing** [comment]
Also similar issues regarding enumerations. Enable statement of Enum parameters are not updated immediately if dependent on other enumarations.

**Comment by casella** [comment]
@niklwors, @sbaetzing do you still have this problem with 1.25.0?

**Comment by abuntrock** [comment]
@casella I will check this and will report back.

**Comment by abuntrock** [comment]
From my point of view, this ticket is not completely closed yet. Here is a minimum example which works independently from the BRSL. I tested this with OMEdit 1.24.3 and the current nightly v1.26.0-dev-55-gadf5d405e1. The behavior is the same in both cases.
```modelica
package ImmediateUpdate
    model ClassWithInstances
  DependingOnString dependingOnString  annotation(
      Placement(transformation(origin = {-30, 30}, extent = {{-10, -10}, {10, 10}})));
  DependingOnEnum dependingOnEnum annotation(
      Placement(transformation(origin = {10, 30}, extent = {{-10, -10}, {10, 10}})));
  DependingOnReal dependingOnReal  annotation(
      Placement(transformation(origin = {50, 30}, extent = {{-10, -10}, {10, 10}})));
  equation

  end ClassWithInstances;

  model DependingOnString
  
    parameter String controller = "Velocity controller" "Controller"
    annotation (choices(
        choice="Torque controller",
        choice="Velocity controller",
        choice="Position controller"));
  parameter Real realParam1=1.00  annotation (HideResult = true, Evaluate = true, Dialog(
    enable=controller =="Position controller"),choices(checkBox=true));
    parameter Real realParam2=1.00  annotation (Evaluate = false, Dialog(
     enable=controlMode > 2));
  
  
  protected
    parameter Integer controlMode = if controller =="Torque controller" then 1
                              else if controller =="Velocity controller" then 2
                              else if controller =="Position controller" then 3
                              else -1 "Control mode";
  
  end DependingOnString;

  model DependingOnEnum
  parameter SomeEnum enumParam = SomeEnum.enum1;
  
  parameter Real realParam1 = 5 annotation(Dialog(enable = enumParam==SomeEnum.enum2));
  parameter Real realParam2 = 5 annotation(Dialog(enable = controlMode==2));

  protected
    parameter Integer controlMode = if enumParam ==SomeEnum.enum1 then 1
                              else if enumParam ==SomeEnum.enum2 then 2
                              else -1 "Control mode";
  
  end DependingOnEnum;

  model DependingOnReal
  
  parameter Real someReal = 5;
  
  parameter Real realParam = 5 annotation(Dialog(enable = someReal>10));

  end DependingOnReal;

  type SomeEnum = enumeration(
      enum1
     "First choice",
      enum2
     "Second choice") "Some enumeration";
end ImmediateUpdate;
```

From the original issue, the point from @adeas31 on 4.4.2023 is still valid.
In the above minimum example this means that the enable visualization of `dependingOnString.realParam2` and `dependingOnEnum.realParam2` in class `ClassWithInstance` are not immediately updated when setting `dependingOnString.controller="Position controller"` and `dependingOnEnum.enumParam=SomeEnum.enum2` respectively.

Additionally, my point raised on 10.7.2024 is also still valid. In the above minimum example this means that the enable visualization of `dependingOnReal.realParam` in class `ClassWithInstance` is not immediately updated when setting `dependingOnReal.someReal=15`.

In all three cases the update only takes places after closing and reopening the parameter dialog.

**Comment by casella** [comment]
@adeas31 do you think you can fix this?

**Comment by casella** [comment]
@adeas31 can this be handled for 1.26.0?

**Comment by adeas31** [comment]
I tested the example model given by @niklwors in https://github.com/OpenModelica/OpenModelica/issues/10145#issuecomment-2778019162 again and `DependingOnString` and `DependingOnEnum` works fine. `DependingOnReal` doesn't work but I guess it is easy to fix.

**Comment by adeas31** [comment]
Please test this with tomorrow's nightly.
If this worked, I will add this to 1.26

**Comment by adeas31** [comment]
Backported this to maintenance/v1.26 with #14828

**Comment by abuntrock** [comment]
Thanks for the fix and the backporting @adeas31.
I tested this with today's nightly and the minimum example form https://github.com/OpenModelica/OpenModelica/issues/10145#issuecomment-2778019162 as well as the examples from our library work perfectly. This makes the affected parameter dialogs from our library much easier to use.

Thanks a lot.

I cannot close the issue because I did not create it initially though.
